core/src/al/miniaudio_context.cpp

```diff
@@ -0,0 +1,369 @@
+#define MINIAUDIO_IMPLEMENTATION
+#include <cubos/core/al/miniaudio_context.hpp>
+#include <cubos/core/log.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+
+using namespace cubos::core::al;
+
+class MiniaudioBuffer : public impl::Buffer
+{
+public:
+    ma_decoder decoder;
+
+    MiniaudioBuffer(const void* data, size_t dataSize)
+    {
+        if (ma_decoder_init_memory(data, dataSize, nullptr, &decoder) != MA_SUCCESS)
+        {
+            CUBOS_ERROR("Failed to initialize Decoder from data");
+        }
+
+        mValid = true;
+    }
+
+    ~MiniaudioBuffer() override
+    {
+        ma_decoder_uninit(&decoder);
+    }
+
+    size_t length() override
+    {
+        ma_uint64 lengthInPCMFrames;
+        ma_result result = ma_decoder_get_length_in_pcm_frames(&decoder, &lengthInPCMFrames);
+
+        if (result != MA_SUCCESS)
+        {
+            CUBOS_ERROR("Failed to get the length of audio in PCM frames.");
+            return 0;
+        }
+
+        // Calculate the length in seconds: Length in PCM frames divided by the sample rate.
+        return static_cast<size_t>(lengthInPCMFrames) / decoder.outputSampleRate;
```

Comment:
Better cast to float, otherwise all sub-second sounds are going to have a length of 0.